schema/denm/denm_schema_2-1-0.json

```diff
@@ -0,0 +1,579 @@
+                  "minimum": 0,
+                  "maximum": 65535
+                }
+              }
+            },
+            "detection_time": {
+              "type": "integer",
+              "description": " Time at which the event is detected by the originating ITS-S. For the DENM repetition, this DE shall remain unchanged.",
+              "minimum": 0,
+              "maximum": 4398046511103,
+              "$comment": "Remains the same on repetition; updated if any value changed in the payload, meaning sequence_number remains unchanged but anything else has been updated",
+              "$comment": "Unit: millisecond since ETSI epoch (2004/01/01, so 1072915200000); utcStartOf2004(0), oneMillisecAfterUTCStartOf2004(1)"
+            },
+            "reference_time": {
+              "type": "integer",
+              "description": "Time at which a new DENM, an update DENM or a cancellation DENM is generated.",
+              "minimum": 0,
+              "maximum": 4398046511103,
+              "$comment": "Unit: millisecond since ETSI epoch (2004/01/01, so 1072915200000). utcStartOf2004(0), oneMillisecAfterUTCStartOf2004(1)",
+              "$comment": "Different for each message, even for repetition"
+            },
+            "termination": {
+              "type": "integer",
+              "description": "isCancellation(0), isNegation (1)",
+              "minimum": 0,
+              "maximum": 1
+            },
+            "event_position": {
+              "type": "object",
+              "required": [
+                "latitude",
+                "longitude",
+                "altitude",
+                "confidence"
+              ],
+              "properties": {
+                "latitude": {
+                  "type": "integer",
+                  "description": "Unit: 0.1 microdegree. oneMicrodegreeNorth (10), oneMicrodegreeSouth (-10), unavailable(900000001)",
+                  "default": 900000001,
+                  "minimum": -900000000,
+                  "maximum": 900000001
+                },
+                "longitude": {
+                  "type": "integer",
+                  "description": "Unit: 0.1 microdegree. oneMicrodegreeEast (10), oneMicrodegreeWest (-10), unavailable(1800000001)",
+                  "default": 1800000001,
+                  "minimum": -1800000000,
+                  "maximum": 1800000001
+                },
+                "altitude": {
+                  "type": "integer",
+                  "description": "Unit: 0.01 meter. referenceEllipsoidSurface(0), oneCentimeter(1), unavailable(800001)",
+                  "default": 800001,
+                  "minimum": -100000,
+                  "maximum": 800001
+                },
+                "confidence": {
+                  "$comment": "mandatory on ETSI specification, FIXME?",
```

Comment:
I agree that in our case, it might be preferable to keep this confidence object optional... To be discussed in grooming?

Reply:
Yes it can be something that we can see all together
Yet as every field in the confidence ellipse object (and altitude) is provided an `unavailable` value, being mandatory is not a real issue
It just requires to set it by default when we create the message

Reply:
Agreed!